src/main/java/com/elastic/support/BaseInputs.java

```diff
@@ -2,57 +2,147 @@
+    protected StringInputReader standardStringReader = ResourceCache.textIO.newStringInputReader()
+            .withMinLength(0)
+            .withInputTrimming(true);
+    protected BooleanInputReader standardBooleanReader = ResourceCache.textIO.newBooleanInputReader();
+    protected StringInputReader  standardPasswordReader = ResourceCache.textIO.newStringInputReader()
+            .withInputMasking(true)
+            .withInputTrimming(true)
+            .withMinLength(0);
+    protected StringInputReader standardFileReader = ResourceCache.textIO.newStringInputReader()
+            .withInputTrimming(true)
+            .withValueChecker((String val, String propname) -> validateFile(val));
+    // Input Readers
 
-    public void parseInputs(String[] args){
+    public abstract boolean runInteractive();
+
+    public List<String> parseInputs(String[] args){
         logger.info("Processing diagnosticInputs...");
         jCommander = new JCommander(this);
         jCommander.setCaseSensitiveOptions(true);
         jCommander.parse(args);
+        // If we're in help just shut down.
+        if (help) {
+            jCommander.usage();
+            return emptyList;
+        }
+
+        return ObjectUtils.defaultIfNull(validateDir(outputDir), emptyList);
+
     }
 
-    @Parameter(names = {"-?", "--help"}, description = "Help contents.", help = true)
-    protected boolean help;
-    public boolean isHelp() {
-        return help;
+    public void usage(){
+        if(jCommander != null){
+            jCommander.usage();
+        }
+        else{
+            logger.info("Please rerun with the --help option or with no arguments for interactive mode.");
+        }
     }
-    public void setHelp(boolean help) {
-        this.help = help;
+
+    protected void runOutputDirInteractive(){
+        outputDir = ResourceCache.textIO.newStringInputReader()
+                .withValueChecker(( String val, String propname) -> validateFile(val))
+                .read(SystemProperties.lineSeparator + outputDirDescription);
     }
 
-    @Parameter(names = {"-o", "--out", "--output", "--outputDir"}, description = "Fully qualified path to output directory.")
-    // If no output directory was specified default to the working directory
-    protected String outputDir = SystemProperties.userDir;
-    public String getOutputDir() {
-        return outputDir;
+    public List<String> validatePort(int val){
+        if (val < 1 || val > 65535){
+            return Collections.singletonList("Outside the valid range of port values. 1-65535 ");
+        }
+        return null;
     }
-    public void setOutputDir(String outputDir) {
-        this.outputDir = outputDir;
+
+    public List<String> validateFile(String val) {
+        if (StringUtils.isEmpty(val.trim())) {
+            return null;
+        }
+
+        File file = new File(val);
+
+        if (!file.exists()) {
+            return Collections.singletonList("Specified file could not be located.");
```

Comment:
Probably worth putting the input value with `""` or `[]` wrapping it so that it's easier to debug (e.g., on Windows where paths sometimes have spaces, so if they forgot to use quotes it would be a little more apparent here).

Same would probably help for all of the input errors, but especially for the file ones.

Reply:
If they forget to use quotes the command line parsing utility I use will error out because it will look like a dangling argument that wasn't configured. And if they do put the quotes in there the input library will preserve them. IOW if you put in C:\Users\Greg Nieman\output it won't work. " C:\Users\Greg Nieman\output" _will_ work, and the quotes will carry over into the variable.

But given how the error will occur, it probably warrants emphasizing this in the doc changes I'm working on, even if I am skeptical anyone will pay any attention.